### PR TITLE
Fixes for test conflicts

### PR DESCRIPTION
**Pull request by k1o0**
I've fixed the conflicts between the database tests and they all pass on my laptop.  Frustratingly, these very same tests fail on ZTEST.  I looked at `dat_test` and it passes when I step through or set breakpoints (even conditional ones) but fails otherwise (but not always).  I have a feeling that `save` is asynchronous and so on faster machines it loads the paths before the custom paths file has finished saving.  I'm not sure what the best way of getting around this is.  Let me know how these tests run on the rigs.

### CODE REVIEW

**Comment by jaib1** [comment]
> I have a feeling that save is asynchronous and so on faster machines it loads the paths before the custom paths file has finished saving

did you try throwing in a `pause` to test this theory? not an ideal solution, but if it works we could use it for now

**Comment by k1o0** [comment]
Okay, all tests pass on ZTEST now, however there is a mysterious Alyx issue where the database sometimes returns a 500 for reasons we can't work out, however I think I've ironed out all of the client-side issues with the tests.

File: tests/AlyxPanel_test.m [hunk 1]
```diff
@@ -87,11 +110,13 @@ function setupPanel(testCase)
       % MControl using this as a panel.
       testCase.SubjectUI.addlistener('SelectionChanged', ...
         @(src, evt)testCase.Panel.dispWaterReq(src, evt));
-      
+            
       % Set Alyx Instance and log in
       testCase.Panel.login('test_user', 'TapetesBloc18');
       testCase.fatalAssertTrue(testCase.Panel.AlyxInstance.IsLoggedIn,...
         'Failed to log into Alyx');
+      testCase.fatalAssertEqual(testCase.Panel.AlyxInstance.BaseURL, BaseURLTestSetup,...
```
Comment (jaib1) [inline tests/AlyxPanel_test.m]: @k1o0 I sent you some slack messages about this. This fails.

**Comment by k1o0** [comment]
Should be good to go

**Comment by jaib1** [comment]
```
Undoing your change fixed this and now the url is recorded without needing a test param
```
Hm, so the issue is that `BaseURL` was trying to get passed into two seperate method blocks (`methods (TestClassSetup)`, and `methods (Test)`) which MATLAB 2017B throws a fit over. Should we just ignore this? That means for testing on a rig I'll have to upgrade at least one training rig and one experimental rig to 2018B or newer. 

Also, `test_activeFlag` still fails for me...
```
    Error Details:
    --------------
    Index exceeds the number of array elements (0).
    
    Error in Alyx/set.BaseURL (line 95)
          value = iff(value(1:4)~='http', ['https://' value], value);
    
    Error in Alyx (line 66)
          obj.BaseURL = getOr(p, 'databaseURL', obj.BaseURL);
    
    Error in eui.AlyxPanel (line 74)
                obj.AlyxInstance = Alyx('','');
    
    Error in AlyxPanel_test/test_activeFlag (line 508)
          eui.AlyxPanel(testCase.Figure);
```
The `databaseURL` field for `p = dat.paths` is `''`  so `set.BaseURL` errors when trying to index the value of that field

**Comment by k1o0** [comment]
I don't think MATLAB throws a fit.  If your param is in the setup block and you try to call a test method with it the test runner just removes those tests and throws a warning as you originally pointed out.  What you did to try to fix it was just wrong: the values of properties in the setup and test blocks are independent of one another.  A further error was caused by issue #180.  My laptop was not fast enough to encounter these errors but ZTEST was.  Now all tests appear to work reliably.

**Comment by jaib1** [comment]
```
I don't think MATLAB throws a fit
```
I just checked and this still errors on 2017B...

```
What you did to try to fix it was just wrong: the values of properties in the setup and test blocks are independent of one another
```
I don't follow...I know they're independent of one another. Can you elaborate?

**Comment by k1o0** [comment]
You mean the tests as they are on this commit cause an error on 2017b?  Can you open an issue an attach the command output so I can look at the error?

**Comment by jaib1** [comment]
the issue is exactly as I've said in an above comment. Here is the output - I'm not sure it's worth an issue:

```
>> runtests('AlyxPanel_test')

Error using matlab.unittest.internal.TestSuiteFactory/createSuiteFromParentName (line 27)
Property BaseURL in class AlyxPanel_test must be defined as a TestParameter property for use with method
test_launchSubjectURL.

Error in testsuite>createSuite (line 137)
    suite = factory.createSuiteFromParentName(selector);

Error in testsuite (line 105)
    suites{k} = createSuite(testAsChar, selector, ...

Error in runtests (line 96)
suites = testsuite(parser.Results.tests,pvcell{:});
```
I guess we should always run tests on 2018B or later? And in regards to this 
```
What you did to try to fix it was just wrong: the values of properties in the setup and test blocks are independent of one another
```
I did this in fact _because_ they are independent of another, so I think you may have misunderstood my reason for doing so?

**Comment by k1o0** [comment]
I removed the BaseURL from test_activeFlag (https://github.com/cortex-lab/Rigbox/pull/178/commits/d6b7deff06c363213bcbb299e681a48b7e2d97e3#diff-7edecec6036e5be04c667344842875f5R490) but missed the other two methods.  I didn't realize because it doesn't cause an error on 2018 however the [documentation](https://uk.mathworks.com/help/matlab/matlab_prog/create-advanced-parameterized-test.html#d117e84392) indicates that they shouldn't be accessible to the test methods.  I'll push a fix for this. 

Next time we Skype I'll go over how the AlyxPanel test works and why your fix failed.

**Comment by k1o0** [comment]
Done in https://github.com/cortex-lab/Rigbox/pull/183

**Comment by jaib1** [comment]
```
I removed the BaseURL from test_activeFlag (d6b7def#diff-7edecec6036e5be04c667344842875f5R490) but missed the other two methods. I didn't realize because it doesn't cause an error on 2018 however the documentation indicates that they shouldn't be accessible to the test methods.
```
yeah, this is what I was looking for : )

```
Next time we Skype I'll go over how the AlyxPanel test works and why your fix fail
```
Sure, I just wanted you to be aware that I knew my "fix" wasn't proper

**Pull request merged**